dotCMS/src/main/java/com/dotcms/mock/response/MockHeaderResponse.java

```diff
@@ -0,0 +1,61 @@
+package com.dotcms.mock.response;
+
+import com.dotcms.mock.request.MockHttpRequest;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpServletResponseWrapper;
+
+/**
+ * Mock {@link HttpServletRequest} with a {@link Map} as part of the class that will contain the
+ * Headers.
+ *
+ * See an example here: {@link MockHttpRequest#MockHttpRequest(String, String)}
+ */
+public class MockHeaderResponse extends HttpServletResponseWrapper implements MockResponse {
+  final Map<String, String> headers = new HashMap<String, String>();
+
+  public MockHeaderResponse(HttpServletResponse response) {
+    super(response);
+    if (response.getHeaderNames() != null) {
+      Collection<String> oldHeaders = response.getHeaderNames();
+      oldHeaders.forEach(header->{
+        this.headers.put(header, response.getHeader(header));
+      });
+    }
+  }
+
+  public MockHeaderResponse(HttpServletResponse response, final String key, final String value) {
+    this(response);
+    headers.put(key, value);
+  }
+
+  public HttpServletResponse response() {
+    return this;
+  }
+
+  @Override
+  public String getHeader(String name) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Parameter 'name' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5609581)